### PR TITLE
Replace templates in io/proto_stream.h with upcasting.

### PR DESCRIPTION
**Pull request by pifon2a**

### CODE REVIEW

File: cartographer/io/proto_stream.h [hunk 1]
```diff
@@ -39,8 +40,7 @@ class ProtoStreamWriter {
   ProtoStreamWriter& operator=(const ProtoStreamWriter&) = delete;
 
   // Serializes, compressed and writes the 'proto' to the file.
-  template <typename MessageType>
-  void WriteProto(const MessageType& proto) {
+  void WriteProto(const google::protobuf::MessageLite& proto) {
```
Comment (cschuet) [inline cartographer/io/proto_stream.h]: Let's use Message rather as in other places.

**Comment by pifon2a** [comment]
Done.

**Comment by cschuet** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**